lib/MusicBrainz/Server/Controller/Role/Delete.pm

```diff
@@ -40,22 +40,33 @@ role {
         my $entity_name = $self->{entity_name};
         my $edit_entity = $c->stash->{ $entity_name };
         my $model = $self->{model};
-        if ($model eq 'Area') {
-            $c->stash(
-                is_release_country_area => $c->model('Area')->is_release_country_area($edit_entity->id)
-            );
-        }
+        my $can_delete = $c->model($model)->can_delete($edit_entity->id);
```

Comment:
We were already calling this in line 58 - I was just moving it up here so I can use it under Area too without calling it twice. But I guess it might be fine to just call it twice if you're worried about this causing issues for Genre or Release?

Reply:
OK it was already like that, but if it’s a separate commit it would be more obvious.

It is not causing any issue for Genre, Release, or any other relatable entity type *for now*. It is just fragile code.

Reply:
Ok, I did that.